be/src/io/fs/multi_table_pipe.h

```diff
@@ -74,7 +74,7 @@ class MultiTablePipe : public KafkaConsumerPipe {
     std::atomic<uint64_t> _unplanned_row_cnt {0}; // trigger plan request when exceed threshold
     std::atomic<uint64_t> _inflight_plan_cnt {0}; // how many plan fragment are executing?
     std::atomic<bool> _consume_finished {false};
-    std::shared_ptr<StreamLoadContext> _ctx;
+    StreamLoadContext* _ctx;
```

Comment:
No safe to use row pointer
Is there a cycle reference for StreamLoadContext shared_ptr which causes memory is not deallocated?

Reply:
@airborne12 
Yes it is, for not using weak_ptr to solve the problem:

> Why not use weak_ptr?

> Cuz introducing weak_ptr is not so elegant and has lower performance.
> MultiTablePipe's life cycle is under control of the StreamLoadContext, it is reasonable to use naked ptr to 
> provide MultiTablePipe with an access to the current ctx.